### PR TITLE
Rhmap 4792 - addInputValue to include section details.

### PR DESCRIPTION
**Pull request by witmicko**
# What 
Update the addInputValue function for the $fh.forms client API to include section details.

# JIRA
https://issues.jboss.org/browse/RHMAP-4792

@nialldonnellyfh

### CODE REVIEW

File: src/appforms/src/core/040-Model06Submission.js [hunk 1]
```diff
@@ -952,28 +965,34 @@ appForm.models = function(module) {
...
       var formField = formFields[i];
 
-      if(formField.fieldId._id){
-        if (formField.fieldId._id === fieldId) {
-          return formField;
-        }
-      } else {
-        if (formField.fieldId === fieldId) {
-          return formField;
+        if (formField.fieldId._id) {
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: You can combine this into a single `if` statement

File: src/appforms/src/core/040-Model06Submission.js [hunk 2]
```diff
@@ -952,28 +965,34 @@ appForm.models = function(module) {
       callback(err);
     });
   };
-  Submission.prototype.getInputValueObjectById = function(fieldId) {
+
+  Submission.prototype.getInputValueObjectById = function(fieldId, sectionIndex) {
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: Can you add a comment here.

File: src/appforms/src/core/040-Model06Submission.js [hunk 3]
```diff
@@ -872,19 +885,19 @@ appForm.models = function(module) {
   Submission.prototype.endInputTransaction = function(succeed) {
     this.transactionMode = false;
     var tmpFields = {};
-    var fieldId = "";
+    var fieldIdentifier = "";
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: Can you add a comment on what the `fieldIdentifier` is?

File: src/appforms/src/core/040-Model06Submission.js [hunk 4]
```diff
@@ -816,21 +820,30 @@ appForm.models = function(module) {
...
   Submission.prototype.removeFileValue = function(hashName){
     var subFiles = this.get('filesInSubmission', []);
     if(typeof(hashName) === "string" && subFiles.indexOf(hashName) > -1){
       subFiles.splice(subFiles.indexOf(hashName),1);
       this.set('filesInSubmission', subFiles);
     }
   };
-  Submission.prototype.getInputValueByFieldId = function(fieldId, cb) {
+
+  Submission.prototype.getInputValueByFieldId = function(fieldId, sectionIndex, cb) {
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: Can you add a comment to this function.

**Comment by nialldonnellyfh** [comment]
@witmicko I don't see any tests related to adding values for different section IDs

File: src/appforms/src/core/040-Model06Submission.js (hunk 4 above)
Comment (witmicko) [inline src/appforms/src/core/040-Model06Submission.js]: diff doest load due to size.
https://github.com/witmicko/fh-js-sdk/blob/9adae83b91e0d0a58cfc7d0d069c73486faa895b/src/appforms/tests/tests/core/040-Model06Submission.js#L1020

**Comment by witmicko** [comment]
r

**Comment by witmicko** [comment]
> I don't see any tests related to adding values for different section IDs

@nialldonnellyfh diff doest load due to size.
https://github.com/witmicko/fh-js-sdk/blob/9adae83b91e0d0a58cfc7d0d069c73486faa895b/src/appforms/tests/tests/core/040-Model06Submission.js#L1020

File: Gruntfile.js [hunk 5]
```diff
@@ -19,7 +19,8 @@ module.exports = function(grunt) {
             '!src/appforms/src/core/999*.js',
             '!src/appforms/src/backbone/000*.js',
             '!src/appforms/src/backbone/001*.js',
-            '!src/appforms/src/backbone/999*.js'],
+            '!src/appforms/src/backbone/999*.js'
```
Comment (nialldonnellyfh) [inline Gruntfile.js]: Unnecessary Empty Line

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fh-js-sdk",
-  "version": "2.19.0",
+  "version": "2.19.1-alpha.02",
```
Comment (nialldonnellyfh) [inline package.json]: Probably not a hotfix, I would consider it an added feature so a minor version bump is required.

File: src/appforms/src/core/040-Model06Submission.js [hunk 7]
```diff
@@ -721,7 +721,7 @@ appForm.models = function(module) {
    * 1. onblur (field value)
    * 2. onsubmit (whole submission json)
    *
-   * @param {[type]} params   {"fieldId","value","index":optional}
+   * @param {[type]} params   {"fieldId", "value", "index":optional, "sectionIndex":optional}
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: This should really be

```
@param {object} params   {"fieldId", "value", "index":optional, "sectionIndex":optional}
@param {string}  params.fieldId
@param {number}  params.index
@param {number}  params.sectionIndex
```

File: src/appforms/src/core/040-Model06Submission.js [hunk 8]
```diff
@@ -816,21 +821,36 @@ appForm.models = function(module) {
...
       subFiles.splice(subFiles.indexOf(hashName),1);
       this.set('filesInSubmission', subFiles);
     }
   };
-  Submission.prototype.getInputValueByFieldId = function(fieldId, cb) {
+
+  /**
+   * Returns input value for a field with a given id
+   * @param {string} fieldId - id of the field
+   * @param {number} sectionIndex - optional section id in case field is in repeating section
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: sectionIndex is optional so you can surround with brackets ```@param {number} [sectionIndex]```

File: src/appforms/src/core/040-Model06Submission.js [hunk 9]
```diff
@@ -816,21 +821,36 @@ appForm.models = function(module) {
...
+   * @param cb
```
Comment (nialldonnellyfh) [inline src/appforms/src/core/040-Model06Submission.js]: This is a function param

**Comment by jhellar** [comment]
LGTM, verified following the steps in the JIRA.

**Pull request merged**